### PR TITLE
{bio}[foss/2022a] AFNI v23.0.03 w/ Python 3.10.4 

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

Requires:
- [x] PR #17308

### CODE REVIEW

**Comment by sassy-crick** [comment]
I mark that as Work In Progress as during the installation, some Python packages will be installed in the Python directory, both `lib` and `bin`! That messed up my Python install which I only found by chance. So I need to dig into that more.

**Comment by sassy-crick** [comment]
Hopefully fixed now, so WIP removed

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
eb-rocky9.zao.se - Linux Rocky Linux 9.1 (Blue Onyx), x86_64, AMD Ryzen 9 3900X 12-Core Processor, Python 3.9.14
See https://gist.github.com/29bd405e466e7d85a8f28fcca1f792e0 for a full test report.

File: easybuild/easyconfigs/a/AFNI/AFNI-23.0.03-foss-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+toolchain = {'name': 'foss', 'version': '2022a'}
+toolchainopts = {'openmp': True, 'pic': True}
+
+source_urls = ['https://github.com/afni/afni/archive/']
+sources = ['AFNI_%(version)s.tar.gz']
+checksums = ['d278b8b0eaa707a33120d11018b7ba302484ccf8d34f1b2cf6ea11ff0dec37ed']
+
+builddependencies = [
+    ('CMake', '3.24.3'),
+]
```
Comment (verdurin) [inline easybuild/easyconfigs/a/AFNI/AFNI-23.0.03-foss-2022a.eb:22]: @sassy-crick wondering whether the `buildopts` from the previous versions are still needed?

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 4 out of 4 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux CentOS Linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/2b54f0892a918fa2c228c64271462821 for a full test report.

File: easybuild/easyconfigs/a/AFNI/AFNI-23.0.03-foss-2022a.eb (hunk 1 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/a/AFNI/AFNI-23.0.03-foss-2022a.eb:22]: @verdurin from what I could see it is not required any more:
```
[  0%] Building C object src/CMakeFiles/pcor_objs.dir/configured_pcor/afni_pcor_byte.c.o
cd /dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/src && /apps/sw-eb/software/GCCcore/11.3.0/bin/gcc -DDYNAMIC_suffix=\".so\" -DREPLACE_XT -DDTYPE=byte -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src/niml -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-build/nifti2 -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-src/nifti2 -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-build/znzlib -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-src/znzlib -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-build/nifticdf -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-src/nifticdf -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-build/niftilib -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/_deps/nifti_clib-src/niftilib -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/src/configured_pcor -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src/3DEdge/src -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src/f2c -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/src -I/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src/rickr -O2 -ftree-vectorize -march=native -fno-math-errno -fopenmp -fPIC -DUSE_TRACING -DHAVE_ZLIB -DREAD_WRITE_64 -DLINUX2 -O3 -DNDEBUG -fPIC -MD -MT src/CMakeFiles/pcor_objs.dir/configured_pcor/afni_pcor_byte.c.o -MF CMakeFiles/pcor_objs.dir/configured_pcor/afni_pcor_byte.c.o.d -o CMakeFiles/pcor_objs.dir/configured_pcor/afni_pcor_byte.c.o -c /dev/shm/jsassman/AFNI/23.0.03/foss-2022a/easybuild_obj/src/configured_pcor/afni_pcor_byte.c
```
That said, please feel free to test it to make sure nothing is slipping through. I don't know the program so I might get it wrong.

One thing which I cannot work out is I find two files in my local `PYTHONPATH` which should not be there:
`afnipy.egg-link  easy-install.pth`
The content of them refers to the build directory. Do you or @zao have any idea if they are actually needed?
```
$ cat python3.10/site-packages/afnipy.egg-link
/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src/python_scripts
$ cat python3.10/site-packages/easy-install.pth
/dev/shm/jsassman/AFNI/23.0.03/foss-2022a/afni-AFNI_23.0.03/src/python_scripts
```

Comment (verdurin) [reply easybuild/easyconfigs/a/AFNI/AFNI-23.0.03-foss-2022a.eb:22]: @zao would know much better than I would.

**Comment by tlangfor** [comment]
@sassy-crick Have you fixed the issue of files being installed into Python's directories? I ran into this same issue when trying to build a 2020b version on my system.

**Comment by sassy-crick** [comment]
@tlangfor Apologies for all the delays, busy year.
I had another look at it and hopefully I finally nailed it. The install I tested is running the `afni_system_check.py -check_all` command fine, some warnings messages are due to missing data-directories which, from what I understand, users can download themselves. 
Please try it out and see if that is all working for you now. 
Also, please see the latest PR #19654 as well if that helps.

**Comment by sassy-crick** [comment]
`FFmpeg` downgraded to `4.4.2` so the checks are passed. Apparently there is also the `5.0.1` version. Please advise which one to use. Thanks

**Comment by sassy-crick** [comment]
@boegelbot please test @ generoso

**Comment by sassy-crick** [comment]
Closed as PR #19654 is newer and just got merged.

**Pull request closed without merging**